src/libs/firebase/firestore/featureFlags/useFeatureFlag.ts

```diff
@@ -11,21 +11,30 @@ const appBuildVersion = getAppBuildVersion()
 
 // firestore feature flag documentation:
 // https://www.notion.so/passcultureapp/Feature-Flag-e7b0da7946f64020b8403e3581b4ed42#fff5fb17737240c9996c432117acacd8
-export const useFeatureFlag = (featureFlag: RemoteStoreFeatureFlags): boolean => {
+
+export const useFeatureFlag = (
+  featureFlag: RemoteStoreFeatureFlags | keyof typeof RemoteStoreFeatureFlags
```

Comment:
pourquoi a-t-on besoin des deux types ?

pourquoi pas seulement ?

```suggestion
  featureFlag: RemoteStoreFeatureFlags
```

ou ?

```suggestion
  featureFlag: keyof typeof RemoteStoreFeatureFlags
```

Reply:
Le useFeatureFlag a été utilisé jusqu'aujourd'hui avec l'injection du FF directement en paramètre, l'objectif est de ne pas changer l'utilisation existante et de permettre au développeur d'avoir le choix, si on veux tendre à utiliser que la clé au lieu du FF injecté, je peux faire toutes les modifications de code dans cette PR mais avec un grand nombre de fichiers modifiés

Reply:
je trouve ça mieux de faire le changement jusqu'au bout plutot que d'ajouter de la complexité

mais avant de faire tout les changements, j'aimerais qu'on soit d'accord sur le but de la PR https://github.com/pass-culture/pass-culture-app-native/pull/6499#issuecomment-2144853170